authors/apps/authentication/tests/test_user_registration.py

```diff
@@ -0,0 +1,129 @@
+from django.urls import reverse
+from django.test import TestCase
+from rest_framework.test import APIClient
+import json
+
+from ..models import User
+
+
+class TestQuestionViews(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = {
+            "user": {
+                "username": "Jajcob",
+                "email": "jakk@ejake.jake",
+                "password": "444444444",
+            }
+        }
+        self.user2 = User.objects.create(
+            username="Jajcob", email="jakk@ejake.jake", is_superuser=False
+        )
+
+    def test_register_password_without_capital_letter(self):
+
+        response = self.client.post(
+            "/api/users/",
+            content_type="application/json",
+            data=json.dumps(self.user),
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.data["errors"]["password"][0],
+            "Password should contain at least one Upper Case letter",
+        )
+
+    def test_register_password_without_atleast_one_lower_letter(self):
+
+        self.user["user"]["password"] = "444444R444"
+        response = self.client.post(
+            "/api/users/",
+            content_type="application/json",
+            data=json.dumps(self.user),
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.data["errors"]["password"][0],
+            "Password should contain at least one lower Case letter",
+        )
+
+    def test_register_password_without_non_alphanumeric_password(self):
+
+        self.user["user"]["password"] = "Password"
+
+        response = self.client.post(
+            "/api/users/",
+            content_type="application/json",
+            data=json.dumps(self.user),
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.data["errors"]["password"][0],
+            "Password should be alpha Numerics",
```

Comment:
spelling error: change alpha Numerics to alpha-numeric

Reply:
Thanks, @bisonlou for the highlight, let me correct it